src/V8.rs

```diff
@@ -173,16 +175,29 @@ pub fn get_version() -> &'static str {
   c_str.to_str().unwrap()
 }
 
-// TODO: V8::InitializePlatform does not actually take a UniquePtr but rather
-// a raw pointer. This means that the Platform object is not released when
-// V8::ShutdownPlatform is called.
 /// Sets the v8::Platform to use. This should be invoked before V8 is
 /// initialized.
-pub fn initialize_platform(platform: UniqueRef<Platform>) {
+pub fn initialize_platform(platform: Pin<Arc<Platform>>) {
   let mut global_state_guard = GLOBAL_STATE.lock().unwrap();
+  let mut global_platform_guard = GLOBAL_PLATFORM.lock().unwrap();
   assert_eq!(*global_state_guard, Uninitialized);
-  unsafe { v8__V8__InitializePlatform(platform.into_raw()) };
+
+  {
+    let unpinned_arc = unsafe { Pin::into_inner_unchecked(platform.clone()) };
+    unsafe { v8__V8__InitializePlatform(Arc::as_ptr(&unpinned_arc) as *mut _) };
+  }
+
   *global_state_guard = PlatformInitialized;
+  *global_platform_guard = Some(platform);
+}
+
+/// Sets the v8::Platform to use. This should be invoked before V8 is
+/// initialized.
+pub fn get_current_platform() -> Pin<Arc<Platform>> {
```

Comment:
I'm wondering if `Pin<Arc<Platform>>` could be simplified? @piscisaureus what do you think?

Reply:
I don't think `Pin<Arc<_>>` is necessarily inappropriate, but it doesn't look like the reference count will ever exceed 1.

Reply:
I think the counter does exceed 1, when returnining the Arc, it is closed and this the count is incremented.

Reply:
Also, Arc allows to call Drop on platform once the V8::global_state_guard is called and no other references (Arc) exist anymore. (Fixes this comment: https://github.com/denoland/rusty_v8/pull/706/files#diff-5968ad1a2c8f5b1977d1dc3776ad078298f49e52a8a7bc137ecb46d6a45fc8a9L176)